frontend/modules/viewer/viewer.sagas.ts

```diff
@@ -47,6 +55,28 @@ export function* mapInitialise({surveyPoints, sources = []}) {
 	}
 }
 
+export function* initialiseToolbar() {
+	try {
+		Viewer.on(VIEWER_EVENTS.UPDATE_NUM_CLIP, (clipNumber) => {
+			const isClipEdit = selectIsClipEdit(getState());
```

Comment:
@dleszcz Please movie this to new action and the just dispatch it here. ;)
Because of that simple change, you will be able to avoid unnecessary uses of `getState()`. It's just a migration helper, so after the migration, we will have to remove all `getState()` occurrences.